dc/wireshark/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+mod buffer;
+mod dissect;
+mod field;
+#[cfg(not(test))]
+mod plugin;
+mod value;
+/// This wraps the underlying sys APIs in structures that support a cfg(test) mode that doesn't rely on Wireshark.
+mod wireshark;
+
+/// These are bindgen-generated bindings from bindgen 4.2.5.
+/// Allow warnings since we don't control the bindgen generation process enough for warnings to be worthwhile to fix.
+#[allow(warnings)]
+mod wireshark_sys {
+    #[cfg(test)]
+    #[path = "minimal.rs"]
+    mod sys_impl;
+
+    #[cfg(not(test))]
+    #[path = "full.rs"]
+    mod sys_impl;
```

Comment:
Can we add a comment about the difference between minimal vs. full? Or maybe in generate-bindings.rs? It's not entirely clear to me why cfg(test) lets us pull in a subset of this, I'd expect tests to not bring in any externs (at least I think I had that working)

Reply:
Yeah I can add a comment.

I was still seeing linker errors on macos, even with the flags, but maybe that went away with other changes? Let me see.

Reply:
Ah nope. I get linker errors when running `cargo bolero test` as well.